cpp/build/support.mk

```diff
@@ -96,9 +107,15 @@ else
 pkgconfigdir ?= $(shell pkg-config --variable pc_path pkg-config | awk '{split($$0,a,":"); print a[1]}')
 endif
 
+ifeq ($(BITBAKE_ENV),1)
+sysconfdir := $(PREFIX)/etc/openzwave/
+includedir := $(PREFIX)/include/openzwave/
+docdir := $(PREFIX)/share/doc/openzwave-$(VERSION).$(VERSION_REV)
+else
 sysconfdir ?= $(PREFIX)/etc/openzwave/
```

Comment:
On 16-04-27 02:08 AM, Justin Hammond wrote:
>
> In cpp/build/support.mk 
> <https://github.com/OpenZWave/open-zwave/pull/861#discussion_r61208238>:
>
> > @@ -96,9 +107,15 @@ else
> >  pkgconfigdir ?= $(shell pkg-config --variable pc_path pkg-config | awk '{split($$0,a,":"); print a[1]}')
> >  endif
> >
> > +ifeq ($(BITBAKE_ENV),1)
> > +sysconfdir := $(PREFIX)/etc/openzwave/
> > +includedir := $(PREFIX)/include/openzwave/
> > +docdir := $(PREFIX)/share/doc/openzwave-$(VERSION).$(VERSION_REV)
> > +else
> >  sysconfdir ?= $(PREFIX)/etc/openzwave/
>
> why can't PREFIX be specified during the first run?
>
Bin:
Bitbake used by yocto project is to build embedded linux for small 
devices. It typically does both host build first then doing cross 
compile for the target CPU after, then packaging the final image for the 
target device.
The bitbake used in Yocoto project doesn't allow user to specify the 
PREFIX freely. It is automatically invoking a lot of python / shell 
scripts doing host native / target cross compile build prcesses.
The recursive make happens all the time.
There is no way to enforce the final PREFIX value in the very beginning 
run. Because the PREFIX value for the final target path would be unknown 
during the very first run.
>
> You do realise that PREFIX is used as part of the source compilation, 
> so if its not consistent, then things could break?
>
> (for instance, we use PREFIX when compiling to specify the default 
> location to search for the device database...
>

Bin:
Fully understand the concern. This is why I introduced BITBAKE_ENV to 
make sure it is for bitbake build system only to search for default 
location, using compile tools and options from bitbake... all accordingly.

Regards
Bin
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/OpenZWave/open-zwave/pull/861/files/8c26a41e9f9f3a4a9484c53194c0b3ca7721db4b#r61208238>
>